mlflow/pyfunc/__init__.py

```diff
@@ -345,8 +346,12 @@ def _enforce_schema(pdf: pandas.DataFrame, input_schema: Schema):
     For column types, we make sure the types match schema or can be safely converted to match the
     input schema.
     """
-    if isinstance(pdf, list):
-        pdf = pandas.DataFrame(pdf)
+    if isinstance(pdf, list) or isinstance(pdf, np.ndarray) or isinstance(pdf, dict):
+        try:
+            pdf = pandas.DataFrame(pdf)
+        except Exception:
+            message = "The input type provided could not be converted into a pandas DataFrame."
```

Comment:
I think we should include that the model contains model signature that suggests DataFrame input.

Reply:
And we should also include error message to provide more details.